packages/runtime/src/runtime/showDialog.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  unwrapWidget,
+  type EnsembleWidget,
+  type ShowDialogAction,
+  ShowDialogOptions,
+} from "@ensembleui/react-framework";
+import { EnsembleRuntime } from "./runtime";
+import { cloneDeep, findKey, get, isObject, set } from "lodash-es";
+
+export type ShowDialogApiProps = {
+  action?: ShowDialogAction;
+  openModal?: (...args: any[]) => void;
+};
+
+export const showDialog = (props?: ShowDialogApiProps): void => {
+  const { action, openModal } = props ?? {};
+  if (!action || !openModal || !action?.widget) {
+    return;
+  }
+
+  const widget = action?.widget;
+  const widgetKey = findKey(widget, isObject) || "";
+  let inlineWidget: EnsembleWidget | undefined;
+
+  if ((get(widget, widgetKey) as Record<string, unknown>)?.inputs) {
+    set(widget, "name", widgetKey);
+    set(widget, "properties", get(widget, widgetKey));
```

Comment:
It's better to clone/create a new object here than modify the widget model inline. It's not clear to me why this requires special handling and can't be unwrapped with the util?